src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -517,7 +518,16 @@ private String setupGlobalSettingFile() throws IOException, InterruptedException
      * @throws AbortException in case of error
      */
     private void settingsFromConfig(String settingsConfigId, FilePath settingsFile) throws AbortException {
-        Config c = Config.getByIdOrNull(settingsConfigId);
+
+        Config c = null;
+        Executor executor = Executor.currentExecutor();
+        if (executor != null) {
+            Queue.Executable currentExecutable = executor.getCurrentExecutable();
+            if (currentExecutable != null) {
+                c = ConfigFiles.getByIdOrNull((Run<?, ?>) currentExecutable, settingsConfigId);

```

Comment:
@alvarolobato this part I'm not sure it works in this case (it does works as a workaround in other plugins I have done the migration) - but unfortunate this plugin does not have any test cases so I can not easily check it.

the best would be to have access to the build/run at this location, but I have no idea how to get it from `StepContext`.